Backend/lists.js

```
@@ -25,15 +24,49 @@ Parse.Cloud.define("newList", async (request) => {
 });
 
 Parse.Cloud.define("createInvitation", async (request) => {
-  let mainQuery;
   const relation = await request.user.get("lists");
-  const list = await relation.query().equalTo("objectId", request.params.listId).find({ useMasterKey: true });
-  if (list.length == 1) {
+  const lists = await relation.query().equalTo("objectId", request.params.listId).find({ useMasterKey: true });
+  if (lists.length == 1) {
       return await new Parse.Object("SharecartInvitation").save({
-        list: list[0]
+        list: lists[0]
       }, { useMasterKey: true });
   }
 },{
   fields: ["listId"],
   requireUser: true
 });
+
+Parse.Cloud.define("joinList", async (request) => {
+  const invitations = await new Parse.Query("SharecartInvitation").equalTo("objectId", request.params.code).find({ useMasterKey: true });
+  if (invitations.length == 1) {
+      const invitation = invitations[0];
+      
+      if (invitation.used) {
+          throw("Invitation already used!");
```

Do you want to throw here? That'll crash the app. It'll be better to return an error code. Ideally we'd define an enum with the possible responses from joinList call.

---

this sends the error message back to the app, not sure if it has any consequence on the server side